### PR TITLE
Application: Rework Command Line Handling

### PR DESCRIPTION
**Pull request by Marukesu**
Let most of command line to be handled locally, only handling the window/tab creation in the primary instance.

Fixes: https://github.com/elementary/files/issues/2305

### CODE REVIEW

File: src/View/Window.vala [hunk 1]
```diff
@@ -757,8 +755,10 @@ public class Files.View.Window : Hdy.ApplicationWindow {
 
     private void add_window (GLib.File location = GLib.File.new_for_path (PF.UserUtils.get_real_user_home ()),
                              ViewMode mode = ViewMode.PREFERRED) {
-
-        marlin_app.create_window (location, real_mode (mode));
+        with (new Window (marlin_app)) {
```
Comment (jeremypw) [inline src/View/Window.vala:758]: I think we need to bump the minimum Vala version to use this syntax?

**Comment by jeremypw** [comment]
Fixes the --help and --version issues completely, and creating a new or new tab works immediately where a tab is blocked e.g. due to disconnected server but if the tab is busy due to loading a large folder then nothing happens until the folder is loaded (and a "Files not responding" dialog may appear). However fixing that is not so easy and can be left for another PR.

File: src/View/Window.vala (hunk 1 above)
Comment (Marukesu) [reply src/View/Window.vala:758]: yeah, it was introduced in vala 0.50, will it be fine to bump it? i can use the normal syntax if it preferred.

Comment (jeremypw) [reply src/View/Window.vala:758]: Yes, should be fine. Version 0.56 is supplied with Horus and we will soon be switching to Gtk4 anyway. We need not worry about compatibility with previous versions of elementary.

Comment (jeremypw) [reply src/View/Window.vala:758]: I shall have to remember that syntax - looks neat.

**Review by jeremypw — APPROVED** [review]
Thanks for doing this!

**Pull request merged**